lib/endpoints/class-wp-rest-request-controller.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+class WP_REST_Request_Controller {
+
+	/**
+	 * Sanitizes, validates, and transforms WP_REST_Request query arguments.
```

Comment:
Maybe we should move validation and sanitization to separate methods?

Actually, we could pluck `validate_schema_property` and `sanitize_schema_property` into this class, and then use them for collection queries or CRUD operations on every request.